client/post-editor/media-modal/detail/detail-preview-videopress.jsx

```diff
@@ -93,7 +95,27 @@ class EditorMediaModalDetailPreviewVideoPress extends Component {
 		}
 	};
 
+	receiveMessage = ( event ) => {
+		if ( event.origin && event.origin !== location.origin ) {
+			return;
+		}
+
+		const data = 'string' === typeof event.data
+			? JSON.parse( event.data )
+			: event.data;
```

Comment:
was it with @gwwar that I recently held a discussion on this? there's something about this construction I don't like (also it's unsafe because `JSON.parse()` throws on invalid syntax) because it assumes that string data is JSON data

I would be far more in favor of something else like this…

```js
const parseEventData = data => {
	try {
		return JSON.parse( data );
	} catch (e) {
		return data;
	}
}

…

const data = parseEventData( event.data );
```